docs/docs/network-api-migration.md

```diff
@@ -0,0 +1,697 @@
+        "containerPort": 80,
+        "protocol": "tcp",
+        "name": "web",
+        "networkNames": ["first"]
+      },
+      {
+        "containerPort": 81,
+        "protocol": "tcp",
+        "name": "admin",
+        "networkNames": ["second"]
+      }
+    ]
+  },
+  "networks": [
+    {
+      "mode": "container",
+      "name": "first"
+    },
+    {
+      "mode": "container",
+      "name": "second"
+    }
+  ]
+}
+</pre>
+</td>
+</tr>
+</tbody>
+</table>
+
+#### App with bridge network
+
+<table class="table">
+  <tbody>
+  <tr>
+  <th>
+    <b>1.4 API</b>
+  </th>
+  <th>
+    <b>1.5 API</b>
+  </th>
+  </tr>
+  <tr>
+    <td>
+      <pre>
+{
+  "id": "bridge-net-old",
+  "cpus": 1,
+  "mem": 128,
+  "instances": 1,
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "nginx",
+      "network": "BRIDGE",
+      "portMappings": [
+        {
+          "containerPort": 80,
+          “hostPort”: 0,
+          "protocol": "tcp",
+          "name": "web"        }
+      ]
+    }
+  }
+}
+</pre>
+</td>
+
+<td>
+ <pre>
+ {
+  "id": "bridge-net-new",
+  "cpus": 1,
+  "mem": 128,
+  "instances": 1,
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "nginx"
+    },
+    "portMappings": [
+      {
+        "containerPort": 80,
+        "hostPort": 0,
+        "protocol": "tcp",
+        "name": "web"
+      }
+    ]
+  },
+  "networks": [
+    {
+      "mode": "container/bridge"
+    }
+  ]
+}
+</pre>
+</td>
+</tr>
+</tbody>
+</table>
+
+#### App with host network
+
+<table class="table">
+  <tbody>
+  <tr>
+  <th>
+    <b>1.4 API</b>
+  </th>
+  <th>
+    <b>1.5 API</b>
+  </th>
+  </tr>
+  <tr>
+    <td>
+      <pre>
+{
+  "id": "host-net-old",
+  "cpus": 1,
+  "mem": 128,
+  "instances": 1,
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "nginx",
+      "network": "HOST"
+    }
+  },
+  "portDefinitions": [
+    {
+      "port": 0,
+      "protocol": "tcp",
+      "name": "web"
+    }
+  ]
+}
+</pre>
+</td>
+
+<td>
+ <pre>
+ {
+  "id": "host-net-new",
+  "cpus": 1,
+  "mem": 128,
+  "instances": 1,
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "nginx"
+    }
+  },
+  "portDefinitions": [
+    {
+      "port": 0,
+      "protocol": "tcp",
+      "name": "web"
+    }
+  ],
+  "networks": [
+    {
+      "mode": "host"
+    }
+  ]
+}
+</pre>
+</td>
+</tr>
+</tbody>
+</table>
+
+### Invalid Definitions
+
+#### Mixed Networks (Invalid)
+You cannot combine network types.
+
+<table class="table">
+  <tbody>
+  <tr>
+  <th>
+    <b>1.5 API</b>
+  </th>
+  </tr>
+  <tr>
+    <td>
+      <pre>
+{
+  "id": "x-mixed-networks",
+  "cpus": 1,
+  "mem": 128,
+  "instances": 1,
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "nginx"
+    }
+  },
+  "portDefinitions": [
+    {
+      "port": 0,
+      "protocol": "tcp",
+      "name": "web"
+    }
+  ],
+  "networks": [ // Invalid because network modes must be identical
```

Comment:
"Invalid because network modes must be identical" -- yes, and no. Multiple host networks are not allowed. Multiple container/bridge networks are not allowed. Only multiple container networks are allowed.

Perhaps a better comment here is "Network modes may not be mixed. Only multiple `container` networks are allowed"